k9mail-library/src/main/java/com/fsck/k9/mail/transport/smtp/SmtpTransport.java

```diff
@@ -766,26 +766,24 @@ private void readPipelinedResponse(Queue<String> pipelinedCommands) throws IOExc
         String responseLine = null;
         List<String> results = new ArrayList<>();
         while (noOfPipelinedResponse > 0) {
+            noOfPipelinedResponse--;
             results.clear();
             responseLine = readCommandResponseLine(results);
             try {
                 responseLineToCommandResponse(responseLine, results);
 
             } catch (NegativeSmtpReplyException exception) {
+                if (noOfPipelinedResponse == 0) {
+                    throw exception;
```

Comment:
So we can just get rid of the NegativeSmtpReplyException case, since it's a subclass of MessagingException? :)